pkg/validation/internal/multiarch.go

```diff
@@ -205,58 +218,123 @@ func (data *multiArchValidator) loadInfraLabelsFromCSV() {
 				foundManager = true
-				break
 			}
+
+			// Collect nodeAffinity boundaries for all images
+			data.imageNodeAffinity[c.Image] = append(extractNodeAffinityPlatforms(v.Spec.Template.Spec))
 		}
+
 		// If we do not find a container called manager then we
 		// will add all from the Deployment Specs which is not the
 		// kube-rbac-proxy image scaffold by default
 		if !foundManager {
 			for _, c := range v.Spec.Template.Spec.Containers {
-				if c.Name != default_container_scaffold_by_sdk && len(data.managerImages[c.Image]) == 0 {
-					data.managerImages[c.Image] = append(data.managerImages[c.Image], platform{})
-					data.managerImagesString = append(data.managerImagesString, c.Image)
+				// Skip kube-rbac-proxy or already added images
+				_, exists := data.managerImages[c.Image]
+				if c.Name == default_container_scaffold_by_sdk || exists {
+					continue
 				}
+
+				data.managerImages[c.Image] = make([]platform, 0)
+				data.managerImagesString = append(data.managerImagesString, c.Image)
 			}
 		}
 	}
 
-	data.allOtherImages = make(map[string][]platform)
+	data.otherCSVDeploymentImages = make(map[string][]platform)
 	if data.bundle.CSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs != nil {
 		for _, v := range data.bundle.CSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
 			for _, c := range v.Spec.Template.Spec.Containers {
-				// If not be from manager the add
-				if len(data.managerImages[c.Image]) == 0 {
-					data.allOtherImages[c.Image] = append(data.allOtherImages[c.Image], platform{})
+				// Skip images in the manager image list
+				_, exists := data.managerImages[c.Image]
+				if exists {
+					continue
 				}
+
+				data.otherCSVDeploymentImages[c.Image] = make([]platform, 0)
 			}
 		}
 	}
 
+	data.relatedImages = make(map[string][]platform)
 	for _, v := range data.bundle.CSV.Spec.RelatedImages {
-		data.allOtherImages[v.Image] = append(data.allOtherImages[v.Image], platform{})
+		data.relatedImages[v.Image] = make([]platform, 0)
+	}
+}
+
+// extractNodeAffinityPlatforms scans the deployment spec for
+// affinity.nodeAffinity.requiredDuringSchedulingIngoredDringExecution.nodeSelectorTerms
+// that set platform requirements for kubernetes.io/arch and kubernetes.io/os labels
+func extractNodeAffinityPlatforms(spec corev1.PodSpec) []platform {
+	var platforms = make([]platform, 0)
+	if spec.Affinity == nil ||
+		spec.Affinity.NodeAffinity == nil ||
+		spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil ||
+		spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms == nil {
+		// No platforms set
+		return platforms
+	}
+
+	var terms = spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
+	for _, t := range terms {
+		var arches = make([]string, 0)
+		var os = make([]string, 0)
+
+		for _, e := range t.MatchExpressions {
+			if e.Operator != "In" {
```

Comment:
This won't support anti-affinity, which I think is fine given the goal of this change.

Reply:
The multiarch validator is used in the operator-framework/audit tool to produce a report on the health of operators and is also available for self-validation for operator-sdk authors. It will not block operator publishing. :)